src/Network/Http.php

```diff
@@ -278,13 +277,11 @@ public function send()
          * Set request data
          */
         if ($this->requestData) {
-            $requestDataQuery = http_build_query($this->requestData, '', $this->argumentSeparator);
-
             if (in_array($this->method, [self::METHOD_POST, self::METHOD_PATCH, self::METHOD_PUT])) {
-                curl_setopt($curl, CURLOPT_POSTFIELDS, $requestDataQuery);
+                curl_setopt($curl, CURLOPT_POSTFIELDS, $this->getRequestData());
             }
             elseif ($this->method == self::METHOD_GET) {
-                curl_setopt($curl, CURLOPT_URL, $this->url . '?' . $requestDataQuery);
+                curl_setopt($curl, CURLOPT_URL, $this->url . '?' . $this->getRequestData());
```

Comment:
@JaxWilko Sorry for the delay in getting back to you on this PR. Just one thing to note - this line here for GET requests should only accept request data provided through the proper method calls - it should not allow `CURLOPT_POSTFIELDS` to provide data, however, your new `getRequestData` method can potentially allow POSTFIELDS data through if the `requestData` array is empty.